### ISSUE TITLE
False positive for `Style/CollectionCompact` on before Ruby3.1 

### ISSUE THREAD

**Issue by kyoshidajp** [rubocop/rubocop#11037]
## Expected behavior

No offenses.

## Actual behavior

`Style/CollectionCompact` is detected.

## Steps to reproduce the problem

Before Ruby3.1.

```ruby
[1, nil].to_enum.reject(&:nil?)
```

Run RuboCop.

```
Inspecting 1 file
C

Offenses:

sample.rb:1:18: C: [Correctable] Style/CollectionCompact: Use compact instead of reject(&:nil?).
[1, nil].to_enum.reject(&:nil?)
                 ^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

According to [Feature \#17312](https://bugs.ruby-lang.org/issues/17312), `Enumerable` and `Enumerable::Lazy` classes have `compact` after [this PR](https://github.com/ruby/ruby/pull/3851). It looks like after Ruby3.1.

## RuboCop version

```
% bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.6.6) [x86_64-darwin20]
```
